src/firewall/core/rich.py

```diff
@@ -655,16 +670,29 @@ def check(self):
                 raise FirewallError(errors.INVALID_RULE, "forward-port and action")
 
         # source-port
-        elif type(self.element) == Rich_SourcePort:
-            if not functions.check_port(self.element.port):
-                raise FirewallError(errors.INVALID_PORT, self.element.port)
-            if self.element.protocol not in [ "tcp", "udp", "sctp", "dccp" ]:
+        elif type(self.element) == Rich_SourcePort:	
+            if not functions.check_port(self.element.port):	
+                raise FirewallError(errors.INVALID_PORT, self.element.port)	
+            if self.element.protocol not in [ "tcp", "udp", "sctp", "dccp" ]:	
                 raise FirewallError(errors.INVALID_PROTOCOL, self.element.protocol)
-
-        # other element and not empty?
-        elif self.element is not None:
-            raise FirewallError(errors.INVALID_RULE, "Unknown element %s" % 
-                                type(self.element))
+        
+        # tcp-mss-clamp
+        elif type(self.element) == Rich_Tcp_Mss_Clamp:
+            if str(self.element.value):
```

Comment:
The XML attribute value should always be a string. Using `str()` in this area is unnecessary.